### PR TITLE
Refactor FXIOS-5506 [v110] move theme into shared

### PR DESCRIPTION
**Pull request by OrlaM**
[FXIOS-5506](https://mozilla-hub.atlassian.net/browse/FXIOS-5506)
#12816 

This is a prerequisite to using the theme manager in extensions.

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -10535,7 +10534,6 @@
 				C8E2E80E23D20FD2005AACE6 /* FxAWebViewController.swift in Sources */,
 				E14F7DF2288F3F9F00E3722C /* ThemedTableSectionHeaderFooterView.swift in Sources */,
 				E1442FBF294782B6003680B0 /* CGRect+Extension.swift in Sources */,
-				8AB8574427D963960075C173 /* Notifiable.swift in Sources */,
```
Comment (nbhasin2) [inline Client.xcodeproj/project.pbxproj:10538]: Notifiable still exists in utils but is gone from pbxproj? 

Am I reading this right or only got removed from parts of pbxproj 🤔

Comment (nbhasin2) [reply Client.xcodeproj/project.pbxproj:10538]: I cloned the project and this is more of a cleanup right?

Comment (OrlaM) [reply Client.xcodeproj/project.pbxproj:10538]: It should be gone completely from the project and now lives in BrowserKit.

File: Client/AccountSyncHandler.swift [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 import Foundation
 import Storage
+import Common
```
Comment (nbhasin2) [inline Client/AccountSyncHandler.swift:7]: Is common replacing shared in the future or they will mostly live together to serve a different purpose?

File: Client.xcodeproj/project.pbxproj (hunk 1 above)
Comment (nbhasin2) [reply Client.xcodeproj/project.pbxproj:10538]: image

Perfect, had to cleanup XCode but have the one and only instance

File: Shared/Theme/ThemeManager.swift [hunk 3]
```diff
@@ -63,41 +66,37 @@ final class DefaultThemeManager: ThemeManager, Notifiable {
...
 
         // overwrite the user interface style on the window attached to our scene
         // once we have multiple scenes we need to update all of them
-        UIWindow.keyWindow?.overrideUserInterfaceStyle = currentTheme.type.getInterfaceStyle()
-
+        if #available(iOS 15, *) {
+            appDelegate?.window??.windowScene?.keyWindow?.overrideUserInterfaceStyle = currentTheme.type.getInterfaceStyle()
+        }
         // old way of updating the theme. We should pass in and save sceneDelegate instead of appDelegate
         appDelegate?.window??.overrideUserInterfaceStyle = currentTheme.type.getInterfaceStyle()
```
Comment (adudenamedruby) [inline Shared/Theme/ThemeManager.swift:83]: Should this be in an else? Or is it fine this way?

**Review by adudenamedruby — APPROVED** [review]
LGTM aside from the one comment.

File: Client/AccountSyncHandler.swift (hunk 2 above)
Comment (OrlaM) [reply Client/AccountSyncHandler.swift:7]: It's possible eventually there will be no more use for shared but it's likely a very long time down the road.

**Review by nbhasin2 — APPROVED** [review]
LGTM

File: Shared/Theme/ThemeManager.swift (hunk 3 above)
Comment (OrlaM) [reply Shared/Theme/ThemeManager.swift:83]: It's fine to call both. I will clean this part up when we move to multi-window support.

**Pull request merged**